nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -720,19 +720,28 @@ public TransferResult<Nullness, NullnessStore> visitFieldAccess(
     Symbol symbol = ASTHelpers.getSymbol(fieldAccessNode.getTree());
     setReceiverNonnull(updates, fieldAccessNode.getReceiver(), symbol);
     Nullness nullness = NULLABLE;
-    boolean fieldMayBeNull =
-        NullabilityUtil.mayBeNullFieldFromType(symbol, config, getCodeAnnotationInfo(state));
-    if (handler
-        .onDataflowVisitFieldAccess(
-            fieldAccessNode,
-            symbol,
-            state.getTypes(),
-            state.context,
-            apContext,
-            values(input),
-            updates)
-        .equals(NullnessHint.HINT_NULLABLE)) {
-      fieldMayBeNull = true;
+    boolean fieldMayBeNull;
+    switch (handler.onDataflowVisitFieldAccess(
```

Comment:
Weird to call the handler before the base algorithm, but nothing in this call depends on `mayBeNullFieldFromType`

Reply:
Yeah, from an optimization standpoint I don't know which call is more likely to make `fieldMayBeNull==true`, which we would ideally do first.  But let's not worry about that for now.

Reply:
The problem is, `onDataflowVisitFieldAccess(...)` can reverse the decision of `mayBeNullFieldFromType(...)` no matter what the decision actually is, so we can't flip the order here and get short-circuiting (at least not without reducing the scope of the Handler callback to just returning a boolean rather than a `NullnessHint`, which we could for now, but not sure it's worth it).